[FIX] sale: remove warnings on product type change

jeep-odoo opened the pull request:
Before the commit:
Previously, when the product type of an item in a sales order was changed, a
warning would be displayed, but users were still allowed to make the change.
Although there are other checks in place to prevent changes to the product type
under certain conditions, the general warnings were unnecessary.

After the commit:
Now, there will be no general warning when changing a product's type.

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/181221.png)](https://mergebot.odoo.com/odoo/odoo/pull/181221)

kcv-odoo commented:
Does not seems right fix :thought_balloon:

jeep-odoo commented:
I have added a second commit where I removed both warnings as discussed with @dit-odoo . She will test it functionally first. That's why I also removed the "Request for review". I will add it back once we come to a conclusion.

jeep-odoo commented:
After testing functionality, @dit-odoo suggested removing the warnings since users were still able to change the product type despite the warnings.

Feyensv commented:
Honestly, I don't see the problems of having a warning here :thinking: 
Could be improve from 'cannot' to 'shouldn't' but doesn't seem like something that should be removed, nor something that is related to 18 :thinking: 

cc: @dalaOdoo

jeep-odoo commented:
> Honestly, I don't see the problems of having a warning here 🤔 Could be improve from 'cannot' to 'shouldn't' but doesn't seem like something that should be removed, nor something that is related to 18 🤔
> 
> cc: @dalaOdoo

I agree that "shouldn't" would be a better fit here, but Dipali (DIT) wanted me to remove those warnings completely. As the task was mentioned on the [pad](https://pad.odoo.com/p/testSales18), I did it for 18.

dit-odoo commented:
> > Honestly, I don't see the problems of having a warning here 🤔 Could be improve from 'cannot' to 'shouldn't' but doesn't seem like something that should be removed, nor something that is related to 18 🤔
> > cc: @dalaOdoo
> 
> I agree that "shouldn't" would be a better fit here, but Dipali (DIT) wanted me to remove those warnings completely. As the task was mentioned on the [pad](https://pad.odoo.com/p/testSales18), I did it for 18.

Feyensv commented:
Talked about it with DIT ~2 weeks ago and we concluded that those warnings are still meaningful and rarely appear all together (though they do indeed appear together on the first product of the demo data).
Even if this onchange warning is not blocking, it is useful to discourage users as most users are not aware of the implications of this change.
